examples/step-36/step-36.cc

```diff
@@ -472,51 +471,19 @@ namespace Step36
 // @sect3{The <code>main</code> function}
 int main(int argc, char **argv)
 {
-  try
-    {
-      using namespace dealii;
-      using namespace Step36;
-
-      Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  using namespace dealii;
+  using namespace Step36;
 
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
 
-      // This program can only be run in serial. Otherwise, throw an exception.
-      AssertThrow(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD) == 1,
-                  ExcMessage(
-                    "This program can only be run in serial, use ./step-36"));
 
-      EigenvalueProblem<2> problem("step-36.prm");
-      problem.run();
-    }
+  // This program can only be run in serial. Otherwise, throw an exception.
+  AssertThrow(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD) == 1,
+              ExcMessage(
+                "This program can only be run in serial, use ./step-36"));
 
-  // All the while, we are watching out if any exceptions should have been
-  // generated. If that is so, we panic...
-  catch (std::exception &exc)
-    {
-      std::cerr << std::endl
-                << std::endl
-                << "----------------------------------------------------"
-                << std::endl;
-      std::cerr << "Exception on processing: " << std::endl
-                << exc.what() << std::endl
-                << "Aborting!" << std::endl
-                << "----------------------------------------------------"
-                << std::endl;
-
-      return 1;
-    }
-  catch (...)
-    {
-      std::cerr << std::endl
-                << std::endl
-                << "----------------------------------------------------"
-                << std::endl;
-      std::cerr << "Unknown exception!" << std::endl
-                << "Aborting!" << std::endl
-                << "----------------------------------------------------"
-                << std::endl;
-      return 1;
-    }
+  EigenvalueProblem<2> problem("step-36.prm");
+  problem.run();
```

Comment:
One more thing: from the discussion in #6784 we should keep this so that MSVC can produce nice output.

Reply:
ah well, that's a bummer. Personally I would consider it MSVC issue and get rid of those lines of code throughout. But I will put them back in here.